src/storage/src/hummock/file_cache/manager.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::PathBuf;
+
+use nix::sys::statfs::{statfs, FsType as NixFsType, EXT4_SUPER_MAGIC};
+
+use super::error::{Error, Result};
+
+#[derive(Clone, Copy, Debug)]
+pub enum FsType {
+    Ext4,
+    Xfs,
+}
+
+#[derive(Debug)]
+pub struct FileCacheManagerOptions {
+    pub dir: String,
+}
+
+pub struct FileCacheManager {
+    _dir: String,
```

Comment:
It's not a good pattern in rust that an used variable is prefixed with underscore.